### PR TITLE
ADD: New function - Apply to sweeps

### LINKED ISSUE #204: Issue with cfradial1 writer: ValueError: cannot rename 'sweep_fixed_angle' because it is not a variable or dimension in this dataset
* xradar version: 0.6.4
* Python version: 3.12.5
* Operating System: MacOS Sonoma 14.6.1

```python
import xradar as xd
import matplotlib.pyplot as plt
from open_radar_data import DATASETS

# Fetch a sample radar file
file = DATASETS.fetch('KATX20130717_195021_V06')

# Open the radar file into a DataTree object
dtree = xd.io.open_nexradlevel2_datatree(file)

xd.io.to_cfradial1(dtree, "out.nc")
```
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/var/folders/yq/7n9vj93d6bq4gq64hx_6qgym0000gp/T/ipykernel_81303/1809764418.py in ?()
----> 1 xd.io.to_cfradial1(dtree)

~/Downloads/Others/xradar/xradar/io/export/cfradial1.py in ?(dtree, filename, calibs)
    287         The name of the output netCDF file.
    288     calibs: Bool, optional
    289         calibration parameters
    290     """
--> 291     dataset = _variable_mapper(dtree)
    292 
    293     # Check if radar_parameters, radar_calibration, and
    294     # georeferencing_correction exist in dtree

~/Downloads/Others/xradar/xradar/io/export/cfradial1.py in ?(dtree, dim0)
    101         Dataset containing mapped radar variables.
    102     """
    103 
    104     sweep_info = _sweep_info_mapper(dtree)
--> 105     vol_info = _main_info_mapper(dtree).drop_vars("fixed_angle")
    106     sweep_datasets = []
    107     for grp in dtree.groups:
    108         if "sweep" in grp:

~/Downloads/Others/xradar/xradar/io/export/cfradial1.py in ?(dtree)
     78     """
     79     dataset = (
     80         dtree.root.to_dataset()
     81         .drop_vars("sweep_group_name", errors="ignore")
---> 82         .rename({"sweep_fixed_angle": "fixed_angle"})
     83     )
     84     return dataset

~/miniconda3/envs/xradar-dev/lib/python3.12/site-packages/xarray/core/dataset.py in ?(self, name_dict, **names)
   4358         Dataset.rename_vars
   4359         Dataset.rename_dims
   4360         DataArray.rename
   4361         """
-> 4362         return self._rename(name_dict=name_dict, **names)

~/miniconda3/envs/xradar-dev/lib/python3.12/site-packages/xarray/core/dataset.py in ?(self, name_dict, **names)
   4296         """
   4297         name_dict = either_dict_or_kwargs(name_dict, names, "rename")
   4298         for k in name_dict.keys():
   4299             if k not in self and k not in self.dims:
-> 4300                 raise ValueError(
   4301                     f"cannot rename {k!r} because it is not a "
   4302                     "variable or dimension in this dataset"
   4303                 )

ValueError: cannot rename 'sweep_fixed_angle' because it is not a variable or dimension in this dataset

```

### PR DESCRIPTION
**Pull request by syedhamidali**
- [x] Closes #204 #208 
- [x] Tests added
- [x] Changes are documented in `history.md`

**Description:**

This PR introduces a new utility function, `apply_to_sweeps`, that allows users to apply a custom function to each sweep within a radar volume stored in a `DataTree` object. This is particularly useful for tasks where you need to perform the same operation across multiple sweeps, such as calculating derived quantities like rain rate from radar reflectivity.

**Example Usage:**

Here’s an example of how to use the `apply_to_sweeps` function to calculate rain rates using the Marshall-Palmer power law:

```python
import xradar as xd
from open_radar_data import DATASETS

# Fetch a sample radar file
file = DATASETS.fetch('KATX20130717_195021_V06')

# Open the radar file into a DataTree object
dtree = xd.io.open_nexradlevel2_datatree(file)

# Define the function to calculate rain rate using the Marshall-Palmer relation
def xradar_rain_rate(ds):
    def _rain_rate(dbz, a=200.0, b=1.6):
        """Calculates the inverse of input decibel values
        Convert to rain rate using the Marshall-Palmer relation
        """
        Z = 10.0 ** (dbz / 10.0)  # Convert dBZ to linear Z
        return (Z / a) ** (1.0 / b)  # Apply the Marshall-Palmer power law

    # Apply the rain rate calculation to the DBZH field
    ds['RAIN_RATE'] = _rain_rate(ds['DBZH'])
    ds['RAIN_RATE'].attrs = {'unit': 'mm/h',
                             'long_name': 'Rain Rate'}
    return ds

# Apply the rain rate function to all sweeps in the DataTree
dtree = xd.util.apply_to_sweeps(dtree, xradar_rain_rate)

# Check that the rain_rate variable has been added to one of the sweeps
for var in dtree['sweep_3']:
     print(var)
```
```python
DBZH
VRADH
WRADH
sweep_mode
sweep_number
prt_mode
follow_mode
sweep_fixed_angle
RAIN_RATE
```

Here's another example for example using `filter_radar` function

```python
file = DATASETS.fetch('RAW_NA_000_125_20080411183259')
dtree = xd.io.open_iris_datatree(file)

def filter_radar(ds):
    ds['DBZH_F'] = ds.where((ds['DBZH']>10) & (ds['DBZH']<60))['DBZH']
    return ds

dtree = xd.util.apply_to_sweeps(dtree, filter_radar)
dtree = dtree.xradar.georeference()

fig, ax = plt.subplots(1, 2, figsize=(10, 4))
dtree['sweep_0']['DBZH'].plot(x='x', y='y', vmin=-10, vmax=60, cmap='Spectral_r', ax=ax[0])
dtree['sweep_0']['DBZH_F'].plot(x='x', y='y', vmin=-10, vmax=60, cmap='Spectral_r', ax=ax[1])
```
image

```python
# If you want to apply this filter on all other fields
def filter_radar(ds):
    ds = ds.where((ds['DBZH']>10) & (ds['DBZH']<60))
    return ds
```

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@syedhamidali Great idea!

We already have implemented the georeferencing to work over a tree. We might think to generalize that approach using this Pull Request.

For the georeferencing we used accessors and the `.pipe` method under the hood.

From my perspective it would make sense to implement your approach as accessor, too.

The API would be like this:

```python
dtree = dtree.xradar.apply(function, *args, **kwargs)
```

Where `apply` would be your `apply_to_sweep` and `function` the function to be applied. If done the right way with the accessors the function can be applied to DataArrays, Datasets and DataTrees.

Does that make sense?

**Comment by syedhamidali** [comment]
@kmuehlbauer Thanks!

Yeah, I agree that `dtree = dtree.xradar.apply(function, *args, **kwargs)` would be a great enhancement. It would make the approach more versatile and consistent with the existing API. However, I’m not yet familiar with how these accessors are built, so I’d appreciate any guidance or examples on how to implement this. Or, if you have the time to implement it yourself, that would be great too!

**Comment by kmuehlbauer** [comment]
@syedhamidali I'll respond with some more details later this weekend. Thanks for being supportive of the overall idea!

**Comment by syedhamidali** [comment]
@kmuehlbauer, when you get a chance, could you take a quick look at PR #203? I’d really appreciate your feedback.

File: docs/history.md [hunk 1]
```diff
@@ -1,14 +1,20 @@
 # History
 
+## Development Version
+
+* ADD: Auto_Read function to read the data automatically using `xd.io.read` ({pull})`205` by [@syedhamidali](https://github.com/syedhamidali).
```
Comment (mgrover1) [inline docs/history.md:5]: This section should only include changes proposed in this pull request - please remove the other PRs notes

**Review by mgrover1 — CHANGES_REQUESTED** [review]
I like this approach - to @kmuehlbauer 's point, we can follow up with that in another PR, I think this functionality will be useful! Please address the comment related to only including the updates related to this PR

File: docs/history.md (hunk 1 above)
Comment (syedhamidali) [reply docs/history.md]: Done!

**Comment by kmuehlbauer** [comment]
Thanks @syedhamidali!

**Pull request merged**